### DISCUSSION TITLE
frb v2 - Is there a list of reserved words?

### DISCUSSION THREAD

**Discussion by erdemyerebasmaz** [fzyzcjy/flutter_rust_bridge#1889] [General]
Hello,
Recently started migrating to v2 and have an issue with a field name, `template`.

We have a struct with a field named `template` and I also added an identically named field with various different types on other structs to test it out and they all ended up getting an underscore suffix after generation, becoming `template_`.

My question is, is `template` a reserved word now? If it is, is there a work-around annotation or where can I find a resource for the list of reserved words?

I couldn't find it in:
- [Dart reserved keywords](https://dart.dev/language/keywords)
- [Rust reserved keywords](https://doc.rust-lang.org/reference/keywords.html)

**Comment by erdemyerebasmaz** [comment]
Ah, I may be mistaken but generator seems to respect [C++ reserved keywords](https://www.umsl.edu/~lawtonb/224/oview2a.html) now...

I was able to reproduce with non shared keywords such as friend, goto, register.

**Reply by fzyzcjy** [reply]
Hmm, I suspect it is either a bug in flutter_rust_bridge, or some feature change in e.g. https://pub.dev/packages/ffigen. Feel free to convert this discussion into an issue, and feel free to PR, or alternatively I will fix on next batch!

**Reply by erdemyerebasmaz** [reply]
@fzyzcjy Thanks for the swift response. I was not entirely sure where to open it because I haven't really looked into the internal changes between v1 & v2 yet.

I'll check [ffigen repository](https://github.com/dart-lang/native/tree/main/pkgs/ffigen) to see if there's anything related & convert this into an issue.

**Reply by fzyzcjy** [reply]
No worries, even if it is dependency feature change, I guess we need to / can do something in this repo to workaround it, so just open issue here :)

**Reply by erdemyerebasmaz** [reply]
Opened the issue:
- https://github.com/fzyzcjy/flutter_rust_bridge/issues/1890
